Bunkum.Protocols.Gemini/Socket/SocketGeminiListener.cs

```diff
@@ -0,0 +1,109 @@
+using System.Net;
+using System.Net.Security;
+using System.Net.Sockets;
+using System.Security.Authentication;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Text.RegularExpressions;
+using Bunkum.Listener;
+using Bunkum.Listener.Extensions;
+using Bunkum.Listener.Protocol;
+using Bunkum.Listener.Request;
+using NotEnoughLogs;
+
+namespace Bunkum.Protocols.Gemini.Socket;
+
+public partial class SocketGeminiListener : BunkumGeminiListener
+{
+    private System.Net.Sockets.Socket? _socket;
+    private readonly Uri _listenEndpoint;
+    private readonly X509Certificate2 _cert;
+
+    [GeneratedRegex("^[a-zA-Z]+$")]
+    private static partial Regex LettersRegex();
+    
+    public SocketGeminiListener(Uri listenEndpoint, Logger logger) : base(logger)
+    {
+        this._listenEndpoint = listenEndpoint;
+        this._cert = new(File.ReadAllBytes("cert.pfx"), "aaaa");
```

Comment:
For obvious reasons, this can't go in master. I think it would be acceptable to have a constructor accepting a `X509Certificate2` directly, but another option is creating a new JSON `Config` and then having that config file contain a file path & key to the certificate. Or both options.